tests/utils/tmp_file.cpp

```diff
@@ -1,20 +1,24 @@
 /*
  * This file is part of covfie, a part of the ACTS project
  *
- * Copyright (c) 2022 CERN
+ * Copyright (c) 2022-2023 CERN
  *
  * This Source Code Form is subject to the terms of the Mozilla Public License,
  * v. 2.0. If a copy of the MPL was not distributed with this file, You can
  * obtain one at http://mozilla.org/MPL/2.0/.
  */
 
-#include <boost/filesystem.hpp>
+// Local include(s).
 #include <tmp_file.hpp>
 
-boost::filesystem::path get_tmp_file()
+// System include(s).
+#include <cstdio>
+
+std::filesystem::path get_tmp_file()
 {
-    return boost::filesystem::temp_directory_path() /
-           boost::filesystem::unique_path(
-               "covfie_test_%%%%_%%%%_%%%%_%%%%.covfie"
-           );
+    char fname[L_tmpnam];
+    char * dummy = std::tmpnam(fname);
+    (void)dummy;
+    return std::filesystem::temp_directory_path() /
+           std::filesystem::path(fname);
```

Comment:
Yeah this is one of the main reasons I used Boost, temporary file handling in `std::filesystem` is awful... But I guess it's a price we can pay for removing the dependency... I'm not against it per se. But if you want to remove Boost completely it will also need to be removed from the benchmarks.

Reply:
Oh, I do hate this. 😦 The Boost version looks a bit more elegant, though that's not any safer either. So I understand why they left this out of the standard implementation. (Not to give people the false sense that using that function would result in safe code.)

As long as Boost is only used for this one purpose in the benchmarks and examples as well, then sure, I can look into that.

Reply:
Boost is used for way more than just for temporary file creation in the benchmarks and the examples. I won't rewrite all of those to Boost-less C\+\+ myself...

For Detray, which is my current objective, that is not really needed anyway.